src/vs/workbench/contrib/chat/common/promptSyntax/service/promptsService.ts

```diff
@@ -199,17 +200,15 @@ export class PromptsService extends Disposable implements IPromptsService {
 			}
 
 			// match each attached file with each glob pattern and
-			// add the instructions file its rule matches the file
+			// add the instructions file if its rule matches the file
 			for (const file of files) {
 				if (match(applyTo, file.fsPath)) {
 					result.push(uri);
-
-					continue;
 				}
 			}
 		}
 
-		return [...new Set(result)];
+		return [...new ResourceSet(result)];
```

Comment:
@legomushroom This is now correct but still quite wasteful. This iterates over `result` to fill in the set and then iterates over the set fill in an array.

* why not have `result` be a `ResourceSet`
* why not return an iterable over an array

Reply:
Yeah I have that change in a different branch already.